src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/KnowledgeBase.java

```diff
@@ -349,10 +349,10 @@ public Date getCreationDateForCause(String id) {
 
     /**
      * Removes the build failure cause of particular build.
-     * @param build the AbstractBuild.
+     * @param build the build.
      * @throws Exception if something in the KnowledgeBase handling goes wrong.
      */
-    public abstract void removeBuildfailurecause(AbstractBuild build) throws Exception;
+    public abstract void removeBuildfailurecause(Run build) throws Exception;
```

Comment:
This will probably break backwards compatibility for other KnowledgeBase implementations.

I think you need to add a new method that takes a Run and deprecate the old one and make both non abstract.
There are an example or two in the gerrit trigger I think.